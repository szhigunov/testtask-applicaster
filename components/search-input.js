import React from 'react';
import {TextInput, StyleSheet} from 'react-native';

export const SearchInput = ({onChangeText, value}) => {
  return (
    <TextInput
      style={styles.input}
      onChangeText={onChangeText}
      value={value}
      placeholder="Search"
    />
  );
};

const styles = StyleSheet.create({
  input: {
    paddingTop: 10,
    paddingBottom: 10,
    marginLeft: 10,
    marginTop: 10,
    marginRight: 10,
    paddingLeft: 10,
    height: 40,
    backgroundColor: '#fff',
  },
});
