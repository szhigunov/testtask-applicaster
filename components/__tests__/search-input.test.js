/**
 * @format
 */

import 'react-native';
import React from 'react';
import {SearchInput} from '../search-input';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('SearchInput renders correctly', () => {
  let root = renderer.create(<SearchInput onChangeText={jest.fn()} value="" />);
  expect(root.toJSON()).toMatchSnapshot();
});
