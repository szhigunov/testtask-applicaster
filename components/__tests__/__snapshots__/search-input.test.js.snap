// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[MockFunction]}
  placeholder="Search"
  rejectResponderTermination={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "height": 40,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingTop": 10,
    }
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;
